plan/builders.go

```diff
@@ -75,10 +75,13 @@ type Builder interface {
 	ProjectRemap(input Rel, remap []int32, exprs ...expr.Expression) (*ProjectRel, error)
 	// Deprecated: Use AggregateColumns(...).Remap() instead.
 	AggregateColumnsRemap(input Rel, remap []int32, measures []AggRelMeasure, groupByCols ...int32) (*AggregateRel, error)
+	// Deprecated: Use Aggregate(...) instead.
 	AggregateColumns(input Rel, measures []AggRelMeasure, groupByCols ...int32) (*AggregateRel, error)
 	// Deprecated: Use AggregateExprs(...).Remap() instead.
 	AggregateExprsRemap(input Rel, remap []int32, measures []AggRelMeasure, groups ...[]expr.Expression) (*AggregateRel, error)
+	// Deprecated: Use Aggregate(...) instead.
 	AggregateExprs(input Rel, measures []AggRelMeasure, groups ...[]expr.Expression) (*AggregateRel, error)
+	Aggregate(input Rel, measures []AggRelMeasure, groupingExpressions []expr.Expression, groupings [][]uint32) (*AggregateRel, error)
```

Comment:
The design of the new grouping expressions allows one to refer to the same grouping expression without requiring the expression to be copied.  Ideally this would be something that could be hidden under the covers.  If it isn't, it is totally valid to duplicate the expression each time (constructing a new grouping id each time).

An API which hid the grouping references implementation details would be ideal and could look like AggregateExprs.

Reply:
In that case, I won't deprecate AggregateExprs and remove the newly added Aggregate()